lib/registration.pm

```diff
@@ -255,17 +270,17 @@ sub fill_in_registration_data {
 
     # Process modules on sle 15
     if (is_sle '15+') {
-        my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
+        my $modules_needle = ("modules-preselected-" . get_required_var('SLE_PRODUCT'));
         if (check_var('BETA', '1')) {
             assert_screen('scc-beta-filter-checkbox');
             send_key('alt-i');
         }
-        elsif (!check_screen($modules_needle, 0)) {
+        elsif (check_screen('scc-beta-filter-checkbox', 0)) {
```

Comment:
I guess we have to leave original condition here. Initial problem is what is stated in the comment. I have created ticket to improve proxy SCC, so we can override beta flag for requested modules, but as of now let's leave it. And yes, I agree that your code looks cleaner, but it won't work for the scenario for which it has been introduced.

Reply:
sorry, I don't follow you here :) but I can put the original condition as it was working, what I understood is that if the checkbox is present in non-beta product is not enough to determine the soft-failure...why is that, that we need to check for the modules in one-shot?